sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -36,11 +36,10 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
       
         <FunctionsEnablePlaceholder Condition="$(FunctionsEnablePlaceholder) == ''">false</FunctionsEnablePlaceholder>
         <FunctionsEnableWorkerIndexing Condition="$(FunctionsEnableWorkerIndexing) == '' Or $(FunctionsEnableWorkerIndexing)">true</FunctionsEnableWorkerIndexing>
-        <FunctionsEnableMetadataSourceGen>$(FunctionsEnableWorkerIndexing)</FunctionsEnableMetadataSourceGen>
-        <FunctionsAutoRegisterGeneratedMetadataProvider>$(FunctionsEnableWorkerIndexing)</FunctionsAutoRegisterGeneratedMetadataProvider>
+        <FunctionsEnableMetadataSourceGen Condition="$(FunctionsEnableMetadataSourceGen) == '' Or $(FunctionsEnableMetadataSourceGen)">true</FunctionsEnableMetadataSourceGen>
```

Comment:
I followed deliberately the same condition pattern already in this file for consistency but generally these conditions don't make a lot of sense. This doesn't protect anyone from setting a value that doesn't evaluate to a boolean. For example you could still set `foo` as a value which then makes the condition blow up. Wouldn't it be better actually have a condition like the following:

```xml
Condition="$(FunctionsEnableMetadataSourceGen) == '' Or $(FunctionsEnableMetadataSourceGen) != 'false'
```

this would then at least set the default value to `true` when it is empty or when an invalid value like `foo` is set.

Reply:
Is this not a breaking change? Anyone who has previously used `FunctionsEnableWorkerIndexing` to disable the metadata source gen + auto registration will be seeing a different behavior with this change.